### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by alandtsang**
Fix memory leak in function sender_body and main_thread

### CODE REVIEW

File: examples/pkt-gen.c [hunk 1]
```diff
@@ -2094,6 +2095,7 @@ main_thread(struct glob_arg *g)
 			tic = timespec2val(&targs[i].tic);
 		if (!timerisset(&toc) || timespec_ge(&targs[i].toc, &t_toc))
 			toc = timespec2val(&targs[i].toc);
+    free(&targs[i]);
```
Comment (vmaffione) [inline examples/pkt-gen.c]: wrong, targs are allocated all together, not cell by cell

File: examples/pkt-gen.c [hunk 2]
```diff
@@ -1317,6 +1317,7 @@ sender_body(void *data)
 quit:
 	/* reset the ``used`` flag. */
 	targ->used = 0;
+  free(targ->nmd);
```
Comment (vmaffione) [inline examples/pkt-gen.c]: wrong, nm_close() must be used

**Comment by vmaffione** [comment]
Problems fixed in the correct way, it will be published in a while

File: examples/pkt-gen.c (hunk 2 above)
Comment (vmaffione) [inline examples/pkt-gen.c]: yes

File: examples/pkt-gen.c (hunk 1 above)
Comment (vmaffione) [inline examples/pkt-gen.c]: yes --> free(targs)

File: examples/pkt-gen.c (hunk 2 above)
Comment (alandtsang) [inline examples/pkt-gen.c]: OK

File: examples/pkt-gen.c (hunk 1 above)
Comment (alandtsang) [inline examples/pkt-gen.c]: Thanks, I got it

**Pull request closed without merging**